crates/vm/levm/src/vm.rs

```diff
@@ -255,6 +256,54 @@ impl VM {
             self.env.refunded_gas,
         );
 
+        if is_precompile(&current_call_frame.code_address) {
+            let precompile_result = execute_precompile(current_call_frame);
+
+            match precompile_result {
+                Ok(output) => {
+                    self.call_frames.push(current_call_frame.clone());
+
+                    return Ok(TransactionReport {
+                        result: TxResult::Success,
+                        new_state: self.cache.clone(),
+                        gas_used: current_call_frame.gas_used.low_u64(),
+                        gas_refunded: 0,
+                        output,
+                        logs: current_call_frame.logs.clone(),
+                        created_address: None,
+                    });
+                }
+                Err(error) => {
+                    if error.is_internal() {
+                        self.call_frames.push(current_call_frame.clone());
+
+                        return Err(error);
+                    }
+
+                    self.call_frames.push(current_call_frame.clone());
+
+                    // Consume all gas
+                    let left_gas = current_call_frame
+                        .gas_limit
+                        .saturating_sub(current_call_frame.gas_used);
+                    current_call_frame.gas_used =
+                        current_call_frame.gas_used.saturating_add(left_gas);
+
+                    self.restore_state(backup_db, backup_substate, backup_refunded_gas);
+
+                    return Ok(TransactionReport {
+                        result: TxResult::Revert(error),
+                        new_state: self.cache.clone(),
+                        gas_used: current_call_frame.gas_used.low_u64(),
```

Comment:
Here we can do `gas_used = gas_limit`

Reply:
Done in [Set gas used just as gas limit](https://github.com/lambdaclass/ethrex/pull/1454/commits/b77275fc2a5a54d00e947ca06e5e73dd07f13c22)